packages/zwave-js/src/lib/commandclass/EntryControlCC.ts

```diff
@@ -0,0 +1,683 @@
+		description: "A list of supported data types",
+	})
+	public get dataTypeSupported(): EntryControlDataTypes[] {
+		return this._dataTypeSupported;
+	}
+
+	private _eventTypeSupported: EntryControlEventTypes[];
+
+	@ccValue()
+	@ccValueMetadata({
+		...ValueMetadata.ReadOnly,
+		type: "number[]",
+		label: "Supported event types",
+		description: "A list of supported event types",
+	})
+	public get eventTypeSupported(): EntryControlEventTypes[] {
+		return this._eventTypeSupported;
+	}
+
+	private _keyCachedSizeMin: number;
+
+	@ccValue()
+	@ccValueMetadata({
+		...ValueMetadata.ReadOnlyUInt8,
+		label: "Minimum supported value of key cache size",
+	})
+	public get keyCachedSizeMin(): number {
+		return this._keyCachedSizeMin;
+	}
+
+	private _keyCachedSizeMax: number;
+
+	@ccValue()
+	@ccValueMetadata({
+		...ValueMetadata.ReadOnlyUInt8,
+		label: "Maximum supported value of key cache size",
+	})
+	public get keyCachedSizeMax(): number {
+		return this._keyCachedSizeMax;
+	}
+
+	private _keyCachedTimeoutMin: number;
+
+	@ccValue()
+	@ccValueMetadata({
+		...ValueMetadata.ReadOnlyUInt8,
+		label: "Minimum supported value of key cache timeout",
+	})
+	public get keyCachedTimeoutMin(): number {
+		return this._keyCachedTimeoutMin;
+	}
+
+	private _keyCachedTimeoutMax: number;
+
+	@ccValue()
+	@ccValueMetadata({
+		...ValueMetadata.ReadOnlyUInt8,
+		label: "Maximum supported value of key cache timeout",
+	})
+	public get keyCachedTimeoutMax(): number {
+		return this._keyCachedTimeoutMax;
+	}
+
+	public toLogEntry(): MessageOrCCLogEntry {
+		return {
+			...super.toLogEntry(),
+			message: {
+				"supported data types": this.dataTypeSupported
+					.map((dt) => EntryControlDataTypes[dt])
+					.toString(),
+				"supported event types": this.eventTypeSupported
+					.map((et) => EntryControlEventTypes[et])
+					.toString(),
+				"min key cached size": this.keyCachedSizeMin,
+				"max key cached size": this.keyCachedSizeMax,
+				"min key cached timeout": this.keyCachedTimeoutMin,
+				"max key cached timeout": this.keyCachedTimeoutMax,
+			},
+		};
+	}
+}
+
+@CCCommand(EntryControlCommand.EventSupportedGet)
+@expectedCCResponse(EntryControlCCEventSupportedReport)
+export class EntryControlCCEventSupportedGet extends EntryControlCC {}
+
+@CCCommand(EntryControlCommand.ConfigurationReport)
+export class EntryControlCCConfigurationReport extends EntryControlCC {
+	public constructor(
+		driver: Driver,
+		options: CommandClassDeserializationOptions,
+	) {
+		super(driver, options);
+
+		validatePayload(this.payload.length >= 2);
+
+		this._keyCachedSize = this.payload[0];
+		this._keyCachedTimeout = this.payload[1];
+		this.persistValues();
+	}
+
+	private _keyCachedSize: number;
+
+	@ccValue()
+	@ccValueMetadata({
+		...ValueMetadata.UInt8,
+		label: "Key cache size",
+		description: "Number of character that must be stored before sending",
+	})
+	public get keyCachedSize(): number {
+		return this._keyCachedSize;
+	}
```

Comment:
```suggestion
	@ccValue()
	@ccValueMetadata({
		...ValueMetadata.UInt8,
		label: "Key cache size",
		description: "Number of character that must be stored before sending",
	})
	public readonly keyCacheSize: number;
```
same for the timeout. The default metadata should contain min/max values, see specs.